src/test/java/org/kiwiproject/registry/eureka/common/EurekaInstanceTest.java

```diff
@@ -74,4 +77,113 @@ void shouldPreferSecurePortOverNotSecure() {
             assertThat(instance.getHealthCheckUrl()).isEqualTo("https://localhost:8083" + ServicePaths.DEFAULT_HEALTHCHECK_PATH);
         }
     }
+
+    @Nested
+    class ToServiceInstance {
+
+        @Test
+        void shouldCreateAServiceInstanceWithEurekaInstanceData() {
+            var eurekaInstance = EurekaInstance.builder()
+                    .app("appId")
+                    .status("UP")
+                    .hostName("localhost")
+                    .ipAddr("127.0.0.1")
+                    .vipAddress("test-service")
+                    .secureVipAddress("test-service")
+                    .port(Map.of("$", 8080, "@enabled", true))
+                    .securePort(Map.of("$", 0, "@enabled", false))
+                    .adminPort(8081)
+                    .homePageUrl("http://localhost:8080/api")
+                    .statusPageUrl("http://localhost:8081/status")
+                    .healthCheckUrl("http://localhost:8081/health")
+                    .metadata(Map.of(
+                            "commitRef", "abcdef",
+                            "description", "some cool service",
+                            "version", "0.1.0"
+                    ))
+                    .build();
+
+            var serviceInstance = eurekaInstance.toServiceInstance();
+
+            assertThat(serviceInstance.getInstanceId()).isEqualTo(eurekaInstance.getHostName());
```

Comment:
For this test, might want to use soft assertions, since it is testing a bunch of disparate properties

Reply:
Done